app/src/main/java/org/commcarehq/aadharuid/ScanResult.java

```diff
@@ -129,7 +145,55 @@ public ScanResult(String input) {
             state = "";
             pc = "";
             dob = "";
+        } else if (rawString.matches("[0-9]*")) {
+            type = QR_CODE_TYPE_SECURE;
+            byte[] msgInBytes = null;
+            try {
+                msgInBytes = decompressByteArray(new BigInteger(rawString).toByteArray());
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+            if (msgInBytes != null) {
+                int[] delimiters = locateDelimiters(msgInBytes);
+                String referenceId = getValueInRange(msgInBytes, delimiters[0] + 1, delimiters[1]);
+                uid = "xxxxxxxx" + referenceId.substring(0, 4);
+                name = getValueInRange(msgInBytes, delimiters[1] + 1, delimiters[2]);
+                dob = getValueInRange(msgInBytes, delimiters[2] + 1, delimiters[3]);
+                yob = dob.substring(dob.length() - 4);
+                gender = getValueInRange(msgInBytes, delimiters[3] + 1, delimiters[4]);
+                co = getValueInRange(msgInBytes, delimiters[4] + 1, delimiters[5]);
+                dist = getValueInRange(msgInBytes, delimiters[5] + 1, delimiters[6]);
+                lm = getValueInRange(msgInBytes, delimiters[6] + 1, delimiters[7]);
+                house = getValueInRange(msgInBytes, delimiters[7] + 1, delimiters[8]);
+                loc = getValueInRange(msgInBytes, delimiters[8] + 1, delimiters[9]);
+                pc = getValueInRange(msgInBytes, delimiters[9] + 1, delimiters[10]);
+                po = getValueInRange(msgInBytes, delimiters[10] + 1, delimiters[11]);
+                state = getValueInRange(msgInBytes, delimiters[11] + 1, delimiters[12]);
+                street = getValueInRange(msgInBytes, delimiters[12] + 1, delimiters[13]);
+                subdist = getValueInRange(msgInBytes, delimiters[13] + 1, delimiters[14]);
+                vtc = getValueInRange(msgInBytes, delimiters[14] + 1, delimiters[15]);
+                statusCode = STATUS_SUCCESS;
+            } else {
+                statusCode = STATUS_PARSE_ERROR;
```

Comment:
One small point of feedback: I think the filtering of "What kind of barcode is this" needs to be distinct from the parse error for the previous feedback you integrated to be helpful.

IE: If the barcode is a string of numbers as line 148 detects, and parsing failed, we want to know "We saw a barcode we thought we could parse (a big block of numbers) and failed to parse it correctly," which is new information for us. I think that's a distinct situation from "We didn't recognize what was in the barcode at all," since presumably we'd want to act on the former somewhat differently.

Reply:
Since `type` is getting populated even if there is an error in parsing, would not the 2 fields `statusCode` and `type` in combination tell that QR code was of format "secure QR code" but we were not able to parse it. Similarly for situation "We didn't recognize what was in the barcode at all" the type would be [set to "unknown" ](https://github.com/dimagi/AadharUID/blob/newScanner/app/src/main/java/org/commcarehq/aadharuid/ScanResult.java#L196).